Library/Formula/rbenv.rb

```diff
@@ -10,16 +10,37 @@ class Rbenv < Formula
   depends_on "ruby-build" => :recommended
 
   def install
-    inreplace "libexec/rbenv", "/usr/local", HOMEBREW_PREFIX
-    prefix.install Dir["*"]
+    inreplace "libexec/rbenv" do |s|
+      s.gsub!('"${BASH_SOURCE%/*}"/../libexec', libexec.to_s, false)
+      if HOMEBREW_PREFIX.to_s != "/usr/local"
+        s.gsub!(":/usr/local/etc/rbenv.d", ":#{HOMEBREW_PREFIX}/etc/rbenv.d\\0")
+      end
+    end
+
+    # Compile optional bash extension. Failure is not critical.
+    if File.exist? "src/configure"
+      Kernel.system "src/configure"
+      Kernel.system "make", "-C" "src"
+    end
```

Comment:
The generated Makefile isn't battle-tested yet and compilation might fail. However, the module is optional and rbenv will keep working. I don't want to use `system` which would abort formula installation.